src/workerd/api/worker-rpc.c++

```diff
@@ -62,50 +122,50 @@ kj::Promise<capnp::Response<rpc::JsRpcTarget::CallResults>> WorkerRpc::sendWorke
   }
 
   auto callResult = builder.send();
-  auto customEventResult = worker->customEvent(kj::mv(event)).attach(kj::mv(worker));
-
-  // If customEvent throws, we'll cancel callResult and propagate the exception. Otherwise, we'll
-  // just wait until callResult finishes.
-  return callResult.exclusiveJoin(customEventResult
-      .then([](auto&&) -> kj::Promise<capnp::Response<rpc::JsRpcTarget::CallResults>> {
-        return kj::NEVER_DONE;
-      }));
+
+  return ioContext.awaitIo(js, kj::mv(callResult),
+      [](jsg::Lock& js, auto result) -> jsg::Value {
+    return jsg::Value(js.v8Isolate, deserializeV8(js, result.getResult().getV8Serialized()));
```

Comment:
nit:

```suggestion
    return js.v8Ref(deserializeV8(js, result.getResult().getV8Serialized());
```

Reply:
This actually doesn't work, unless I explicitly specify the type, i.e. `js.v8Ref<v8::Value>(`. This looks like a bug, `v8Ref()` needs overloads specifically for `JsValue` and friends, or something. Or maybe I should be using `JsRef`?

Anyway, this code is not new, it was just moved from below. I think this should be cleaned up in a separate change.